kuksa_databroker/databroker/src/broker.rs

```diff
@@ -882,6 +882,24 @@ impl<'a, 'b> DatabaseReadAccess<'a, 'b> {
         }
     }
 
+    pub fn get_entries_by_regex(&self, regex: regex::Regex) -> Result<Vec<Entry>, ReadError> {
+        let mut entries: Vec<Entry> = Vec::new();
+        for key in self.db.path_to_id.keys() {
+            if regex.is_match(key) {
+                entries.push(
+                    self.get_entry_by_id(self.db.path_to_id.get(key).unwrap().to_owned())
+                        .unwrap()
+                        .to_owned(),
```

Comment:
As a general rule, there should be no `unwrap()` in databroker. We want to propagate the errors instead of crashing the whole process.

And there is no need to use `unwrap()` here that I can see. Just iterate through the entries directly, instead of the separate `path_to_id` map.